website/docs/guides/migration/versions/upgrading-to-dbt-utils-v1.0.md

```diff
@@ -0,0 +1,116 @@
+- `dbt_utils.current_timestamp()` is replaced by `dbt.current_timestamp()`. Note that Postgres and Snowflake’s implementation of `dbt.current_timestamp()` differs from the old `dbt_utils` one ([full details here](https://github.com/dbt-labs/dbt-utils/pull/597#issuecomment-1231074577)). If you use Postgres or Snowflake and need identical backwards-compatible behaviour, use `dbt.current_timestamp_backcompat()`. This discrepancy will hopefully be reconciled in a future version of dbt Core.
+- All other cross-db macros have moved to the dbt namespace, with no changes necessary other than swapping out `dbt_utils.` for `dbt.`. See [https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros) for the full list.
+    - In your code editor, you can do a global find and replace with regex: `\{\{\s*dbt_utils\.(any_value|bool_or|cast_bool_to_text|concat|dateadd|datediff|date_trunc|escape_single_quotes|except|hash|intersect|last_day|length|listagg|position|replace|right|safe_cast|split_part|string_literal|type_bigint|type_float|type_int|type_numeric|type_string|type_timestamp|type_bigint|type_float|type_int|type_numeric|type_string|type_timestamp|except|intersect|concat|hash|length|position|replace|right|split_part|escape_single_quotes|string_literal|any_value|bool_or|listagg|cast_bool_to_text|safe_cast|dateadd|datediff|date_trunc|last_day)` → `{{ dbt.$1`
+- The `insert_by_period` materialization has been moved to the [experimental-features repo](https://github.com/dbt-labs/dbt-labs-experimental-features/tree/main/insert_by_period). To continue to use it, add this to your packages.yml file:
+
+```yaml
+packages:
+  - git: https://github.com/dbt-labs/dbt-labs-experimental-features
+    subdirectory: insert_by_period
+    revision: XXXX #optional but highly recommended. Provide a full git sha hash, e.g. 1c0bfacc49551b2e67d8579cf8ed459d68546e00. If not provided, uses the current HEAD.
+```
+
+- `safe_add()` only works with a list of arguments; use `{{ dbt_utils.safe_add(['column_1', 'column_2']) }}` instead of varargs `{{ dbt_utils.safe_add('column_1', 'column_2') }}`.
+- Several long-promised deprecations to `deduplicate()` have been applied:
+    - The `group_by` argument is replaced by `partition_by`.
+    - `relation_alias` is removed. If you need an alias, you can pass it directly to the `relation` argument.
+    - `order_by` is now mandatory. Pass a static value like `1` if you don’t care how they are deduplicated.
+- The deprecated `table` argument has been removed from `unpivot()`. Use `relation` instead.
+
+
+## Resolving error messages
+### `'dict object' has no attribute MACRO_NAME`
```

Comment:
suggesting using toggles instead of headers for troubleshooting section. this will also shorten the content section - what do you think?

```suggestion
<details>
  <summary><code>dict object’ has no attribute MACRO_NAME</code></summary>
  <div>
    <div><b>Cause</b>: No macro called <code>MACRO_NAME</code> exists. This is most likely because the macro has moved to the <code>dbt</code> namespace (see above). It could also be because you haven’t run dbt deps or have misspelled a macro’s name.

<b>Resolution</b>: Replace references to <a href=“https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros”>cross-database macros ()</a> with <code>dbt.MACRO_NAME()</code></div>
  </div>
</details>
<details>
    <summary><code>macro ‘dbt_macro__generate_surrogate_key’ takes not more than 1 argument(s)</code>
</summary>
  <div>
    <div><b>Cause</b>: <code>generate_surrogate_key()</code> requires a single argument containing a list of columns, not a set of varargs. 

<b>Resolution</b>: Change to <code>{{ dbt_utils.generate_surrogate_key([‘column_1’, ‘column_2’]) }}</code> - note the square brackets around the first example.
    </div>
    </div>
</details>
<details>
    <summary>The <code>dbt_utils.surrogate_key</code> has been replaced by <code>dbt_utils.generate_surrogate_key</code></summary>
  <div>
    <div><b>Cause</b>: <code>surrogate_key()</code> has been deprecated. 

<b>Resolution</b>: Decide whether you need to enable backwards compatibility (above). Find and replace <code>dbt_utils.surrogate_key</code> with <code>dbt_utils.generate_surrogate_key</code>.
    </div>
  </div>
</details>
<details>
  <summary>macro <code>dbt_macro__test_expression_is_true</code> takes no keyword argument <code>condition</code></summary>
  <div>
    <div><b>Cause</b>: <code>condition</code> has been removed from the <code>expression_is_true</code> test, now that <code>where</code> is available on all tests automatically.

<b>Resolution</b>: Replace <code>condition</code> with <code>where</code>.

</div>
  </div>
</details>
        
<details>
  <summary>No materialization <code>insert_by_period</code> was found for adapter</summary>
  <div>
    <div><b>Cause</b>: <code>insert_by_period</code> has moved to the experimental features repo (see above).

<b>Resolution</b>: Install the package as described above.
</div>
  </div>
</details>        
```